enhancements/siteconfig-extra-manifest-version-independence.md

```diff
@@ -0,0 +1,413 @@
+    - cluster:
+      extraManifestSearchPaths: 
+        - sno-extra-manifest/
+        - custom-manifests/
+      extraManifests:
+        filter:			
+    ```
+
+  
+* Those path conatins below files:
+
+  ```bash
+  extra manifests in the sno-extra-manifest
+
+  - A.yaml
+  - B.yaml
+  - C.yaml 
+  ```
+
+  ```bash
+  extra manifests in the custom-manifests
+
+  - C.yaml
+  - D.yaml
+  - E.yaml 
+  ```
+
+<table>
+<tr>
+<th>
+Scenario
+</th>
+<th>
+Logical Flow
+</th>
+<th>
+Output CR list
+</th>
+</tr>
+
+<tr>
+<td>
+Include all files from the path
+<pre>
+ yaml
+ [
+    extraManifestSearchPaths: 
+      - sno-extra-manifest/
+      - custom-manifests/					
+  ]
+</pre>
+</td>
+
+
+
+<td>
+  <li>When <b>extraManifestSearchPaths</b> is defined all the CRs will be added in site generation</li>
+ <li>In this example, all the CRs from user git will be added</li>
+</td>
+
+<td>
+  [A.yaml B.yaml C.yaml D.yaml E.yaml] <-- C.yaml picked from <b>custom-manifests</b>
+</td>
+</tr>
+
+<tr>
+<td>
+Siteconfig contains both parameter
+<pre>
+ yaml
+ [
+    extraManifestSearchPaths: 
+      - sno-extra-manifest/
+      - custom-manifests/
+    extraManifestPath: custom-manifests/						
+  ]
+</pre>
+</td>
+
+
+<td>
+  <li>Proposed 4.14 behavior will be applied. SiteGen will discard extraManifestPath, it will only add files from <b>extraManifestSearchPaths</b></li>
+
+</td>
+
+<td>
+ [A.yaml B.yaml C.yaml D.yaml E.yaml] <-- C.yaml picked from <b>custom-manifests</b> path
+</td>
+
+</tr>
+<td>
+Exclude some files
+<pre>
+ yaml
+ [
+    extraManifestSearchPaths: 
+      - sno-extra-manifest/
+      - custom-manifests/
+    extraManifests:
+      filter:
+        inclusionDefault: exclude
+        include:
+          - C.yaml
+          - D.yaml					
+ ]
+</pre>
+</td>
+
+
+<td>
+  <li>The behavior of filter will not change</li>
+  <li>if the default is <i>exclude</i>, only listed *.yaml files will be included in CR generation</li>
+</td>
+
+<td>
+ [C.yaml D.yaml] <--C.yaml picked  <b>custom-manifests</b>
+</td>
+
+</tr>
+
+
+<tr>
+<td>
+Include only mentioned files
+<pre>
+ yaml
+  [
+    extraManifestSearchPaths: 
+      - sno-extra-manifest/
+      - custom-manifests/
+    extraManifests:
+      filter:
+        inclusionDefault: include
+        exclude:
+          - C.yaml
+          - D.yaml					
+ ]
+</pre>
+</td>
+
+
+<td>
+  <li>The behavior of filter will not change</li>
+  <li>if the default is <i>include</i>, only listed *.yaml files will be excluded in CR generation</li>
+</td>
+
+<td>
+ [A.yaml B.yaml E.yaml]
+</td>
+
+</table>
+         
+
+### Risks and Mitigations
+
+N/A
+
+## Implementation choices
+
+N/A
+
+### API Extensions
+
+New parameter will be introduced named `extraManifestSearchPaths` which will accept a `list[]`
+
+### Test Cases
+
+* Add tests on the `extraManifestSearchPaths` behavior:
+ 
+  * Single and multiple path can be passed
+  * Generating correct list of CRs from correct files
+  * File name override works as proposed
+
```

Comment:
Another test case:
* Builtin CRs are not applied to cluster if extraManifestSearchPaths is defined in SiteConfig